### ISSUE TITLE
Magento customer register checkout redirects to empty cart

### ISSUE THREAD

**Issue by ctzral** [nexcess/magento-turpentine#1494]
We're running Turpentine 0.7.1 with Magento 1.9.3.6 on Nginx. During checkout everything works fine, apart from when a customer registers during checkout. Logged in/guest checkouts are fine. In this specific case, when the order is completed (customer/order are created fine), the customer is redirected to an empty cart page and is not logged in.

This appears to be because we fail the check at the top of the Mage_Checkout_OnepageController::successAction():
`$session = $this->getOnepage()->getCheckout();
if (!$session->getLastSuccessQuoteId()) {
    $this->_redirect('checkout/cart');
    return;
}`
So we're getting redirected back to the cart as there is no lastSuccessQuoteId in the checkout session.

Digging a bit deeper, this problem occurs when the registering customer is set as logged in as part of the checkout process. From the customer session model:
`   public function setCustomerAsLoggedIn($customer)
    {
        $this->setCustomer($customer);
        $this->renewSession();
        Mage::getSingleton('core/session')->renewFormKey();
        Mage::dispatchEvent('customer_login', array('customer'=>$customer));
        return $this;
    }`
When we comment out $this->renewSession(); then everything works OK. This method eventually calls session_regenerate_id() which we think might be the issue. Is this a Turpentine configuration issue? Our cookie settings look OK. This is only an issue when we're using Turpentine/Varnish.

**Comment by m-overlund** [comment]
I do not believe this is related to turpentine, but Magento patch 9767 v1 and v2.
Try googling this issue - there's a lot of issues like yours - fx http://www.elsner.com/installing-magento-supee-9767-9767-v2-possible-issues-and-their-solutions/

**Comment by ctzral** [comment]
@m-overlund,Thanks for your reply. When we switch off Turpentine/Varnish, everything works fine. Has anyone had any similar issues to this when using Turpentine?

**Comment by pieterjanliekens** [comment]
I've still got this issue as of today. It never got fixed. Without Varnish mine works fine as well. Think it's not related to Varnish.
I'm trying to debug the issue following the backend responses.

It has something to do with varnish discarding/stripping some cookies maybe?
https://ma.ttias.be/varnish-tip-see-cookies-stripped-vcl/

On a sidenote: these crappy, half-ass-working solutions really want to make me ditch Magento...

**Comment by lykciv** [comment]
I had the same problem on Magento 1.14.4.2 and Nexcessnet_Turpentine 0.7.0. Upgrading to version 0.7.5 fixed the problem for us.
